tests/k8st/tests/test_bgp_advert.py

```diff
@@ -129,43 +129,93 @@ def tearDown(self):
     def get_svc_cluster_ip(self, svc, ns):
         return run("kubectl get svc %s -n %s -o json | jq -r .spec.clusterIP" % (svc, ns)).strip()
 
-    def do_external_curl(self, container, hostname):
-        return run("docker exec %s curl --connect-timeout 2 -m 3 %s" % (container, hostname))
+    def do_external_curl(self, container, hostname, succeed=True):
+        out = "failed"
+        try:
+          for i in range(10):
+            cmd = "docker exec %s curl --connect-timeout 2 -m 3 %s" % (container, hostname)
+            out = retry_until_success(run, function_args=[cmd]) if succeed else run(cmd)
+          if not succeed:
+            self.fail(container + " should not be able to consistently access " + hostname)
+        except subprocess.CalledProcessError:
+          if succeed:
+            raise
+        return out
+
+    def assertECMPRoutes(self, dst, via=["10.192.0.3", "10.192.0.4"]):
```

Comment:
golang seeping in here.. should be `assert_ecmp_routes` :)

Reply:
i did this intentionally since all the assert methods in the python `unittest.TestCase` are camelCased, but maybe it's better to stick to the 🐍